### DISCUSSION TITLE
AzureMonitor: Add pagination to resource picker

### DISCUSSION THREAD

**Discussion by joshhunt** [grafana/grafana#41809] [Data Sources]
**What would you like to be added**:

 Azure Resource Graph has limits on the amount of results returned in a single query. The resource picker should support pagination generally so customers with large Azure accounts can find their resource

 - Paginate initial data request (for subscriptions + resource groups)
   - Note: We _may_ want to split up initial requests into seperate (paginated) requests for subscriptions vs resource groups. In any case, we should ensure that all subscriptions are loaded before resource groups are 
 - Paginate resources request when expanding resource groups
 - Take into account when a user has already selected a resource that is not in the first page of subscriptions/resource groups/resources. This should be handle

Question:
 - when do we actually request the additional pages? All at once, or lazilly as the user scrolls?

**Comment by ketul-kothari** [comment]
@joshhunt @andresmgot - Just a suggestion: If you guys can keep subscription picker, resource group picker and resource picker separate - Data of the pickers getting loaded in the order they are mentioned i.e. on subscription selection - populate resource groups,  on resource group selection - populate resources - this will make huge performance impact.

**Reply by andresmgot** [reply]
The goal of this feature is to provide a similar experience than the Azure Monitor Logs in the Azure portal ([docs here](https://docs.microsoft.com/en-us/azure/azure-monitor/logs/data-platform-logs)). But I agree (without much context), there are performance improvements we can achieve here.

**Comment by andresmgot** [comment]
Referencing https://github.com/grafana/grafana/issues/42106 here. When there is a large amount of resources to list (in the order of thousands), the resource picker becomes unusable. Adding pagination is one of the possible solutions.

**Reply by ketul-kothari** [reply]
@joshhunt @andresmgot  Is there any update on this?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
